user/views.py

```diff
@@ -1,3 +1,129 @@
+            email.send()
+
+            return JsonResponse({'message': 'SUCCESS'}, status=201)
+
+        except KeyError:
+            return JsonResponse({'message': 'KEY_ERROR'}, status=400)
+
+        except ValueError:
+            return JsonResponse({'message': 'VALUE_ERROR'}, status=400)
+
+class ActivateView(View):
+    def get(self, request, uidb64, access_token):
+        try:
+            uid      = force_text(urlsafe_base64_decode(uidb64))
+            user     = User.objects.get(pk=uid)
+            user_dic = jwt.decode(access_token, SECRET_KEY, ALGORITHM)
+
+            if user.id == user_dic["user"]:
+                user.is_active = True
+                user.save()
+                return redirect(EMAIL['REDIRECT_PAGE'])
+
+            return JsonResponse({'message': 'AUTH_FAIL'}, status=400)
+
+        except KeyError:
+            return JsonResponse({'message': 'INVALID_KEY'}, status=400)
+
+class SignInView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            if User.objects.filter(email=data['email']).exists():
+                user = User.objects.get(email=data['email'])
+                password = data['password']
+
+                if bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
+                    access_token = jwt.encode({'user': user.id}, SECRET_KEY, ALGORITHM)
+                    return JsonResponse({'message': 'SUCCESS', 'access_token': access_token.decode('utf-8')}, status=200)
+
+                return JsonResponse({'message': 'INVALID_PASSWORD'}, status=401)
+
+            return JsonResponse({'message': 'USER_DOES_NOT_EXIST'}, status=400)
+
+        except KeyError:
+            return  JsonResponse({'message': 'KEY_ERROR'}, status=400)
+
+        except ValueError:
+            return JsonResponse({'message': 'VALUE_ERROR'}, status=400)
+
+class KakaoSignInView(View):
+    def get(self, request):
+        try:
+            access_token   = request.headers.get('Authorization', None)
+            headers        = {'Authorization': f'Bearer {access_token}'}
+            url            = 'https://kapi.kakao.com/v2/user/me'
+            response       = requests.get(url, headers=headers)
+
+            user           = response.json()
+            social_id      = user['kakao_account']['id']
+            email          = user['kakao_account']['email']
+            name           = user['kakao_account']['profile']['nickname'],
+
+            if email:
+                user, _ = User.objects.get_or_create(
+                    email               = email,
+                    name                = name,
+                    social_id           = social_id,
+                    is_location_agreed  = False,
+                    is_promotion_agreed = False,
+                    social_platform = SocialPlatform.objects.get(platform='kakao')
+                )
```

Comment:
`get_or_create()`를 사용할 때, 인수로 `email`, `name`, `is_location_agreed`, `is_promotion_agreed` ..와 같은 값들도 함께 전달할 경우, 만약에 카카오에서 name을 수정해서 "A" -> "B"로 변경되었다면, 해당 유저가 다음에 로그인할 때 어떻게 될까요?

[hint]
```suggestion
                user, created = User.objects.get_or_create(
                    social_id = social_id
                )
                if created:
                    user.email = email,
                    user.name = name,
                    is_loca ..
                
                access_token
```

Reply:
병민님 저번에 이해시켜주셔서 감사합니다! 
이메일 인증 유닛테스트 완료 후 푸시하겠습니당~